lib/pipeline.js

```diff
@@ -147,6 +149,42 @@ class PipelineModel extends BaseModel {
     }
 
     /**
+     * Fetch all secrets that belong to this pipeline
+     * @property secrets
+     * @return Promise
+    */
+    get secrets() {
+        const listConfig = {
+            params: {
+                pipelineId: this.id
+            },
+            paginate: {
+                count: PAGINATE_COUNT,
+                page: PAGINATE_PAGE
```

Comment:
This is going to lock us in a corner.  Not for this PR< but the next one.  We need to iterate until we've loaded them all.